pipeline/transformer/transformer_sierra/src/main/scala/uk/ac/wellcome/platform/transformer/sierra/transformers/SierraItemsOnOrder.scala

```diff
@@ -0,0 +1,182 @@
+
+/** This transformer creates catalogue items that correspond to items that
+  * are "on order" or "awaiting cataloguing" -- which don't have their own
+  * item record yet.
+  *
+  *
+  * To understand how this works, it's useful to understand the ordering
+  * process:
+  *
+  *   1)  A staff member orders a book.  They create a skeleton bib record
+  *       in Sierra, which is linked to an order record with status 'o' ("on order").
+  *       At this point, there are no item records.
+  *
+  *   2)  When the book arrives, it's "received".  The RDATE on the order
+  *       record gets populated, the status is updated to 'a' ("fully paid"),
+  *       and invoice information gets attached to the order record.
+  *
+  *   3)  At some point after that, an item record is created.  This supercedes
+  *       the order record.
+  *
+  * The Sierra documentation for fixed fields on order records is useful reading:
+  * https://documentation.iii.com/sierrahelp/Default.htm#sril/sril_records_fixed_field_types_order.html%3FTocPath%3DSierra%2520Reference%7CHow%2520Innovative%2520Systems%2520Store%2520Information%7CFixed-length%2520Fields%7C_____11
+  *
+  */
+object SierraItemsOnOrder extends Logging {
+  def apply(
+    id: TypedSierraRecordNumber,
+    itemDataMap: Map[SierraItemNumber, SierraItemData],
+    orderDataMap: Map[SierraOrderNumber, SierraOrderData]
+  ): List[Item[IdState.Unidentifiable.type]] =
+    if (itemDataMap.isEmpty) {
+      orderDataMap.toList
+        .filterNot {
+          case (_, orderData) => orderData.suppressed || orderData.deleted
+        }
+        .sortBy { case (id, _) => id.withoutCheckDigit }
+        .flatMap { case (_, orderData) => createItem(id, orderData) }
+    } else {
+      List()
+    }
+
+  private def createItem(
+    id: TypedSierraRecordNumber,
+    order: SierraOrderData): Option[Item[IdState.Unidentifiable.type]] =
+    (
+      getStatus(order),
+      getOrderDate(order),
+      getReceivedDate(order),
+      getCopies(order)) match {
+
+      // status 'o' = "On order"
+      //
+      // We create an item with a message something like "1 copy ordered for Wellcome Collection on 1 Jan 2001"
+      case (Some(status), orderedDate, _, copies) if status == "o" =>
+        Some(
+          Item(
+            title = None,
+            locations = List(
+              PhysicalLocation(
+                locationType = LocationType.OnOrder,
+                label = createOnOrderMessage(orderedDate, copies)
+              )
+            )
+          )
+        )
+
+      // status 'a' = "Fully paid", RDATE = "when we actually received the thing"
+      //
+      // We create an item with a message like "Awaiting cataloguing for Wellcome Collection"
+      // We don't expose the received date publicly (in case an item has been in the queue
+      // for a long time) -- but we do expect it to be there for these records.
+      case (Some(status), _, Some(_), copies) if status == "a" =>
```

Comment:
I know IntelliJ likes it to be like this if the variable would be unused but `Some(_)` is quite hard to understand, could it be either
```scala
case (Some(status), _, receivedDate, copes) if status == "a" && receivedDate.isDefined =>
```
or
```scala
case (Some(status), _, Some(receivedDate), copes) if status == "a" =>
```